gym_electric_motor/visualization/console_printer.py

```diff
@@ -1,23 +1,73 @@
 from ..core import ElectricMotorVisualization
+import numpy as np
 
 
 class ConsolePrinter(ElectricMotorVisualization):
+    """Prints current training values of the environment on the console.
 
-    _limits = None
+    These include environment state and reference values as well as the number of training steps, the rewards
+    and the cumulative reward. It also shows why an episode has terminated (external, constraint violation)
+    """
 
-    def __init__(self, **kwargs):
-        pass
+    def __init__(self, verbose=0, update_freq=1, **__):
+        """
+        Args:
+            verbose(Int):
+                Integer indicating whether and at which frequency the console will be printed.
+                Options:
+                    0: No Printing
+                    1: Printing after an episode has terminated
+                    2: Printing at each step of an episode
+            update_freq(Int):
+                Unsigned Integer setting the update frequency if verbose is 2.
+                It's value n means that each nth step the corresponding output will be printed
+        """
+        self._limits = None
+        self._print_freq = verbose
+        self._update_freq = update_freq
+        self._num_steps = 0
+        self._cum_reward = 0
+        self._violation = False
+        self._episode = 0
+        np.set_printoptions(formatter={'float': '{:9.3f}'.format})
+        self._reset = False
 
-    def reset(self, reference_trajectories=None, *_, **__):
-        pass
+    def reset(self, *_, **__):
+        """Gets called on environment reset. Handles internal value reset and External reset printing"""
+        self._reset = True
 
-    def set_physical_system(self, physical_system):
+    def set_modules(self, physical_system, *_, **__):
+        """Gets the limits of the current physical system for accurate printing"""
         self._limits = physical_system.limits
 
-    def step(self, state, reference, reward, *_, **__):
-        print(
-            f'State {state * self._limits} \n'
-            f'Reference {reference * self._limits}\n'
-            f'Reward {reward}\n'
-            f'\n'
-        )
+    def step(self, k, state, reference, action, reward, done):
+        """Gets called at each step of the environment.
+        Handles per step printing as well es constraint violation printing"""
+        if self._print_freq:

```

Comment:
Misuse of integers for boolean expressions is no best practice.

```suggestion
        if self._print_freq > 0:
```